dataloadgen.go

```diff
@@ -75,28 +86,38 @@ type loaderBatch[KeyT comparable, ValueT any] struct {
 	errors        []error
 	fetchExecuted bool
 	done          chan struct{}
+	contexts      []context.Context
 }
 
 // Load a ValueT by key, batching and caching will be applied automatically
-func (l *Loader[KeyT, ValueT]) Load(key KeyT) (ValueT, error) {
-	return l.LoadThunk(key)()
+func (l *Loader[KeyT, ValueT]) Load(ctx context.Context, key KeyT) (ValueT, error) {
+	loadContext, span := l.tracer.Start(ctx, "LOAD")
+	defer span.End()
+	return l.LoadThunk(loadContext, key)()
 }
 
 // LoadThunk returns a function that when called will block waiting for a ValueT.
 // This method should be used if you want one goroutine to make requests to many
 // different data loaders without blocking until the thunk is called.
-func (l *Loader[KeyT, ValueT]) LoadThunk(key KeyT) func() (ValueT, error) {
+func (l *Loader[KeyT, ValueT]) LoadThunk(ctx context.Context, key KeyT) func() (ValueT, error) {
 	l.mu.Lock()
 	defer l.mu.Unlock()
 	if it, ok := l.thunkCache[key]; ok {
 		return it
 	}
-	l.startBatch()
+
+	if l.batch == nil {
```

Comment:
startBatch already checks this

Reply:
i actually moved this check here while i was reading through it - minor pattern preference, i can reset it though